subsys/bluetooth/mesh/blob_srv.c

```diff
@@ -821,7 +825,25 @@ static int handle_info_get(const struct bt_mesh_model *mod, struct bt_mesh_msg_c
 	net_buf_simple_add_u8(&rsp, BLOB_BLOCK_SIZE_LOG_MIN);
 	net_buf_simple_add_u8(&rsp, BLOB_BLOCK_SIZE_LOG_MAX);
 	net_buf_simple_add_le16(&rsp, CONFIG_BT_MESH_BLOB_CHUNK_COUNT_MAX);
-	net_buf_simple_add_le16(&rsp, BLOB_RX_CHUNK_SIZE);
+
+#if defined(CONFIG_BT_MESH_LOW_POWER)
+	/* If friendship is established, then chunk size is according to friend's queue size.
+	 * Chunk size = (Queue size * Bytes per segment) - Opcode - Chunk Number - 8 byte MIC (max)
+	 */
+	if (bt_mesh.lpn.established && bt_mesh.lpn.queue_size > 0) {
+		uint16_t chunk_size = (bt_mesh.lpn.queue_size * 12) - 11;
```

Comment:
Is it possible to define or fetch these numbers (to avoid magic numbers)? The first part of your comment maps to what is written, but that Opcode - Chunk Number - 8 byte MIC (max) = 11 is not obvious.

Reply:
Works for me, thanks!